### PR TITLE
Add a script for fetching trial scoring in TSV format for YorkAT

### PR DESCRIPTION
**Pull request by kammerer**
A version of Annemarie's script for fetching trial scoring data and formatting it in such a way that it can be used directly in collaborator's association pipeline.

### CODE REVIEW

File: public/client_example/trial_analysis_YorkAT.rb [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+
+STDERR.puts '8. Generating output TSV to STDOUT'
+
+output_string = CSV.generate(col_sep: "\t") do |csv|
+  csv << ["<Trait>"] + trait_descriptors.map{ |td| td['trait']['name'].gsub(/\s+/, '.') }
+  outputs.
+    select do |_, data|
+      trait_descriptors.select { |td|
+        data['trait_scores'].detect { |ts| ts['trait_descriptor_id'] == td['id'] }
```
Comment (Nuanda) [inline public/client_example/trial_analysis_YorkAT.rb]: I assume this is here to exclude PSUs which do not provide full scoring info. Please note this produces an empty output file to trials like this one: https://bip.earlham.ac.uk/trial_scorings/2. But i understand this is intentional.

File: public/client_example/trial_analysis_YorkAT.rb [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      }.
+      count == trait_descriptors.count
```
Comment (Nuanda) [inline public/client_example/trial_analysis_YorkAT.rb]: Please, please change this code ;)! I was thinking for an entire minute where the variable `count` is being defined... Then, I saw the dot. I think having this one local variable assigned in the previous line, and compared here, would yield more readable code.

File: public/client_example/trial_analysis_YorkAT.rb [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    outputs[plant_scoring_unit['scoring_unit_name']]['plant_accession'] = plant_accessions.detect { |pa| pa['id'] == plant_scoring_unit['plant_accession_id']}
+  end
+  page += 1
+end
+
+
+STDERR.puts '8. Generating output TSV to STDOUT'
+
+output_string = CSV.generate(col_sep: "\t") do |csv|
+  csv << ["<Trait>"] + trait_descriptors.map{ |td| td['trait']['name'].gsub(/\s+/, '.') }
```
Comment (Nuanda) [inline public/client_example/trial_analysis_YorkAT.rb]: Why use this specific "<Trait>" column header here? This is the plant line name column, so this kind of name would be preferable, so the users who find the generated tsv file after half a year on their disks do not wonder, what values are actually presented in the 1st column.

If this is not an option (e.g. some external tool requires this particular header), maybe consider a line of comment above the header?

Similar question regarding the strange "." notation for trait names. Is that required by some external tool? If not, perhaps quoting the names would be sufficient?

File: public/client_example/trial_analysis_YorkAT.rb [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+          }
+
+          ts['score_value'] if ts
+        }
+    end
+end
+
+puts output_string
+
+STDERR.puts '6. Finished'
```
Comment (Nuanda) [inline public/client_example/trial_analysis_YorkAT.rb]: It's not "6." anymore :).

**Comment by kammerer** [comment]
I am not sure about #681 but I think Annemarie does not work on it any longer anyway. I would not however delete that branch as of yet because the original version of the script may be useful (although both of them could use some tidying).

File: public/client_example/trial_analysis_YorkAT.rb (hunk 3 above)
Comment (kammerer) [inline public/client_example/trial_analysis_YorkAT.rb]: This is based on an example file I was given by Wiktor. I do not have any more information, so I just tried to make the output as similar as possible. I will add some comments regarding this formatting.

File: public/client_example/trial_analysis_YorkAT.rb (hunk 1 above)
Comment (kammerer) [inline public/client_example/trial_analysis_YorkAT.rb]: Yes, this is intentional, however needs to be confirmed with the collaborator.

File: public/client_example/trial_analysis_YorkAT.rb (hunk 2 above)
Comment (kammerer) [inline public/client_example/trial_analysis_YorkAT.rb]: OK

**Comment by Nuanda** [comment]
:+1:

**Pull request merged**